test/MusicStore.E2ETests/DotnetRunTestRunner.cs

```diff
@@ -0,0 +1,97 @@
+        {
+        }
+
+        public async Task RunTests(
+            ServerType serverType,
+            RuntimeFlavor runtimeFlavor,
+            ApplicationType applicationType,
+            RuntimeArchitecture runtimeArchitecture)
+
+        {
+            var testName = $"DotnetRunTests_{serverType}_{runtimeFlavor}_{applicationType}";
+            using (StartLog(out var loggerFactory, testName))
+            {
+                var logger = loggerFactory.CreateLogger("DotnetRunTests");
+                var musicStoreDbName = DbUtils.GetUniqueName();
+                var applicationPath = Helpers.GetApplicationPath();
+                var deploymentParameters = new DeploymentParameters(
+                    applicationPath, serverType, runtimeFlavor, runtimeArchitecture)
+                {
+                    PublishApplicationBeforeDeployment = false,
+                    TargetFramework = Helpers.GetTargetFramework(runtimeFlavor),
+                    Configuration = Helpers.GetCurrentBuildConfiguration(),
+                    EnvironmentName = "Development",
+                    ApplicationType = applicationType,
+                    UserAdditionalCleanup = parameters =>
+                    {
+                        DbUtils.DropDatabase(musicStoreDbName, logger);
+                    },
+                    EnvironmentVariables =
+                    {
+                        { MusicStoreConfig.ConnectionStringKey, DbUtils.CreateConnectionString(musicStoreDbName) },
+                    },
+                };
+
+                using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, loggerFactory))
+                {
+                    var deploymentResult = await deployer.DeployAsync();
+                    var httpClientHandler = new HttpClientHandler { UseDefaultCredentials = true };
+                    var httpClient = deploymentResult.CreateHttpClient(httpClientHandler);
+
+                    var response = await RetryHelper.RetryRequest(
+                        () => httpClient.GetAsync(string.Empty), logger, deploymentResult.HostShutdownToken);
+
+                    Assert.False(response == null, "Response object is null because the client could not " +
+                        "connect to the server after multiple retries");
+
+                    var validator = new Validator(httpClient, httpClientHandler, logger, deploymentResult);
+
+                    logger.LogInformation("Verifying home page");
+                    // Verify HomePage should validate that we're using precompiled views.
+                    await validator.VerifyHomePage(response);
+
+                    var layoutPath = Path.Combine(applicationPath, "Views", "Shared", "_Layout.cshtml");
+                    var originalContent = File.ReadAllText(layoutPath);
+                    try
+                    {
+                        File.WriteAllText(layoutPath, originalContent + " ", Encoding.UTF8);
```

Comment:
@rynowak think this is a worthwhile scenario to cover?

Reply:
Yes